middle_end/flambda/inlining/inlining_decision.ml

```diff
@@ -137,6 +174,41 @@ module Call_site_decision = struct
     | Recursion_depth_exceeded
     | Never_inline_attribute -> Do_not_inline
     | Inline { attribute = _; unroll_to; } -> Inline { unroll_to; }
+
+
+  let report_reason fmt t =
+    match (t : t) with
+    | Environment_says_never_inline ->
+      Format.fprintf fmt "the@ environment@ says@ never@ to@ inline"
+    | Unrolling_depth_exceeded ->
+      Format.fprintf fmt "the@ maximum@ unrolling@ depth@ has@ been@ exceeded"
+    | Max_inlining_depth_exceeded ->
+      Format.fprintf fmt "the@ maximum@ inlining@ depth@ has@ been@ exceeded"
+    | Recursion_depth_exceeded ->
+      Format.fprintf fmt "the@ maximum@ recursion@ depth@ has@ been@ exceeded"
+    | Never_inline_attribute ->
+      Format.fprintf fmt "the@ call@ has@ an@ attribute@ forbidding@ inlining"
+    | Inline { attribute = None; unroll_to = None; } ->
+      Format.fprintf fmt "the@ decision@ at@ the@ function@ declaration"
+    | Inline { attribute = Some Always; unroll_to = _; } ->
+      Format.fprintf fmt "the@ [@inline always]@ attribute@ on@ the@ call"
+    | Inline { attribute = Some Unroll; unroll_to = Some n; } ->
+      Format.fprintf fmt "the@ [@unroll %d]@ attribute@ on@ the@ call" n
+
+    (* this should not happen *)
+    | Inline { attribute = None; unroll_to = Some _; }
+    | Inline { attribute = Some Unroll; unroll_to = None; }
+      ->
+      Misc.fatal_errorf "This should not happen (Inlining_decision.report is not in sync\
+                         with Inlining_decision.make_decision_for_call_site)"
+
+  let report fmt t =
+    Format.fprintf fmt "@[<v>The function call %s been inlined@ because of @[<hov>%a@]@]"
```

Comment:
The `"of"` part doesn't fit with some of the messages. I don't really like the solution of moving it into the relevant cases of `report_reason`, but if you can't find a better solution that's still an improvement.

Reply:
I've fixed the wording to not need the "of" anymore